### PR TITLE
[core]update FileFormatType from Enum to String, to support userdefin…

### PR DESCRIPTION
**Pull request by YanZhangN**
…e FileFormat.

### Purpose
update FileFormatType from Enum to String, to support userdefine fileformat， besides [orc, avro, parquet].

Linked issue: close #xxx

### CODE REVIEW

File: paimon-common/src/main/java/org/apache/paimon/options/FileFormatType.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+package org.apache.paimon.options;
+
+import org.apache.paimon.options.description.DescribedEnum;
+import org.apache.paimon.options.description.InlineElement;
+
+import static org.apache.paimon.options.description.TextElement.text;
+
+/** Specifies the file format type for store. */
+public class FileFormatType implements DescribedEnum {
```
Comment (JingsongLi) [inline paimon-common/src/main/java/org/apache/paimon/options/FileFormatType.java:27]: I feel that FileFormatType can be directly removed, just use the "orc" and "avro" strings, which are almost stable to the point where they cannot be modified.

Comment (YanZhangN) [reply paimon-common/src/main/java/org/apache/paimon/options/FileFormatType.java]: FileFormatType is removed, use String to represent FileFormatType instead.

File: paimon-common/src/main/java/org/apache/paimon/CoreOptions.java [hunk 2]
```diff
@@ -1157,7 +1161,7 @@ public static Path path(Options options) {
         return new Path(options.get(PATH));
     }
 
-    public FileFormatType formatType() {
+    public String formatType() {
         return options.get(FILE_FORMAT);
```
Comment (JingsongLi) [inline paimon-common/src/main/java/org/apache/paimon/CoreOptions.java:1165]: I think we need to toLowerCase here. Consider:
```
Options options = new Options();
options.setString(CoreOptions.FILE_FORMAT.key(), "AVRO");
System.out.println(options.get(CoreOptions.FILE_FORMAT));
```

Upper case works fine, the change should be compatible to upper case usage.
You can add test in `FileFormatTest`.

Comment (YanZhangN) [reply paimon-common/src/main/java/org/apache/paimon/CoreOptions.java]: fixed, and and some tests in `FileFormatTest `.

**Review by JingsongLi — APPROVED** [review]
+1 Thanks @YanZhangN

**Pull request merged**